src/sdam/server_selection.ts

```diff
@@ -287,13 +289,15 @@ export function readPreferenceServerSelector(readPreference: ReadPreference): Se
       return [];
     }
 
-    if (
-      topologyDescription.type === TopologyType.Single ||
-      topologyDescription.type === TopologyType.Sharded
-    ) {
+    if (topologyDescription.type === TopologyType.Single) {
       return latencyWindowReducer(topologyDescription, servers.filter(knownFilter));
     }
 
+    if (topologyDescription.type === TopologyType.Sharded) {
+      const selectable = servers.length > 0 ? servers : deprioritized;
```

Comment:
But wouldn't `servers` here contain also deprioritized ones? Do we filter out them somewhere before that?

Reply:
Good catch. I've updated to filter them out now.